src/shared_realm.cpp

```diff
@@ -221,6 +222,14 @@ void Realm::open_with_config(const Config& config,
             shared_group = std::make_unique<SharedGroup>(*history, options);
         }
     }
+    catch (realm::FileFormatUpgradeRequired const& ex) {
+        if (config.schema_mode != SchemaMode::ResetFile) {
+            translate_file_exception(config.path, config.read_only());
+            return;
```

Comment:
C++ does not support jumping back to the point of a throw from within a catch handler.